api-admin/src/main/java/bio/terra/pearl/api/admin/controller/i18n/I18nController.java

```diff
@@ -21,17 +22,18 @@ public I18nController(LanguageTextService languageTextService, PortalService por
 
   @Override
   public ResponseEntity<Object> listLanguageTexts(String language, String portalShortcode) {
-    Portal portal =
-        portalService
-            .findOneByShortcode(portalShortcode)
-            .orElseThrow(() -> new NotFoundException("Portal not found"));
+    Optional<Portal> portal = portalService.findOneByShortcode(portalShortcode);
+
+    // default to English if no language is provided
+    String lang = Objects.requireNonNullElse(language, "en");
 
     HashMap<String, String> languageTexts;
-    if (language != null) {
-      languageTexts = languageTextService.getLanguageTextMapForLanguage(portal.getId(), language);
+    // if the portal is not found or not specified, at least return the global/system language texts
+    // it's possible to need language texts without a loaded portal context in the admin tool
```

Comment:
we need to load language texts a little bit earlier now, so we don't necessarily have a portal context yet. hence this slight refactor